proxy/src/outbound.rs

```diff
@@ -54,19 +55,25 @@ where
     B: tower_h2::Body + 'static,
 {
     type Request = http::Request<B>;
-    type Response = http::Response<telemetry::sensor::http::ResponseBody<tower_h2::RecvBody>>;
+    type Response = http::Response<telemetry::sensor::http::ResponseBody<transparency::HttpBody>>;
     type Error = Error;
-    type Key = FullyQualifiedAuthority;
+    type Key = (FullyQualifiedAuthority, Protocol);
     type RouteError = ();
     type Service = Buffer<Balance<Discovery<B>>>;
 
     fn recognize(&self, req: &Self::Request) -> Option<Self::Key> {
-        req.uri().authority_part().map(|authority|
-            FullyQualifiedAuthority::new(
+        req.uri().authority_part().map(|authority| {
+            let auth = FullyQualifiedAuthority::new(
                 authority,
                 self.default_namespace.as_ref().map(|s| s.as_ref()),
-                self.default_zone.as_ref().map(|s| s.as_ref()))
-        )
+                self.default_zone.as_ref().map(|s| s.as_ref()));
+
+            let proto = match req.version() {
```

Comment:
TIOLI: replace with `impl From<http::Version> for Protocol`?

Reply:
Yea, could be done, but as this is only ever done in one place currently, having the match inline feels more descriptive to me.